router/core/router.go

```diff
@@ -751,6 +745,172 @@ func (r *Router) bootstrap(ctx context.Context) error {
 		})
 	}
 
+	s3Providers := map[string]config.S3StorageProvider{}
+	cdnProviders := map[string]config.CDNStorageProvider{}
+
+	for _, provider := range r.storageProviders.S3 {
+		if _, ok := s3Providers[provider.ID]; ok {
+			return fmt.Errorf("duplicate s3 storage provider with id '%s'", provider.ID)
+		}
+		s3Providers[provider.ID] = provider
+	}
+
+	for _, provider := range r.storageProviders.CDN {
+		if _, ok := cdnProviders[provider.ID]; ok {
+			return fmt.Errorf("duplicate cdn storage provider with id '%s'", provider.ID)
+		}
+		cdnProviders[provider.ID] = provider
+	}
+
+	if r.graphApiToken != "" {
+
+		var client persistedoperation.Client
+
+		if provider, ok := cdnProviders[r.persistedOperationsConfig.Storage.ProviderID]; ok {
+
+			c, err := cdn.NewClient(provider.URL, r.graphApiToken, cdn.Options{
+				Logger:        r.logger,
+				TraceProvider: r.tracerProvider,
+			})
+			if err != nil {
+				return err
+			}
+			client = c
+
+			r.logger.Info("Fetching persisted operations from CDN",
+				zap.String("providerID", provider.ID),
+			)
+		} else if provider, ok := s3Providers[r.persistedOperationsConfig.Storage.ProviderID]; ok {
```

Comment:
Can you rewrite this without else if, e.g. using functions?

Reply:
Please elaborate.